src/proc_cpuview.c

```diff
@@ -937,6 +966,28 @@ int proc_cpuinfo_read(char *buf, size_t size, off_t offset,
 		return total_len;
 	}
 
+	if (get_host_personality(&host_personality) < 0)
+		return log_error(0, "Failed to get host personality\n");
+
+	if (get_task_personality(fc->pid, &caller_personality) < 0)
+		return log_error(0, "Failed to get caller process (pid: %d) personality", fc->pid);
+
+	/* do we need to change thread personality? */
+	if (host_personality != caller_personality) {
+		int ret;
+
+		ret = personality(caller_personality);
+		if (ret == -1) {
+			return log_error(0, "Call to personality(%d) failed: %s\n",
+				caller_personality, strerror(errno));
+		}
+
+		lxcfs_debug("task (tid: %d) personality was changed %d -> %d\n",
+				(int)syscall(SYS_gettid), ret, caller_personality);
+
+		personality_changed = true;
+	}
```

Comment:
So I'm not completely sure but I think this doesn't work. We can't change the personality past this function lest we confuse subsequent callers. The thread is a worker thread so doesn't go away. IOW, at the end of this function you need to restore the original personality. One possibility is to wrap `proc_cpuinfo_read()` so am like:

```diff
diff --git a/src/proc_fuse.c b/src/proc_fuse.c
index 5416d43..2e6bc09 100644
--- a/src/proc_fuse.c
+++ b/src/proc_fuse.c
@@ -1513,8 +1513,11 @@ __lxcfs_fuse_ops int proc_read(const char *path, char *buf, size_t size,
                return read_file_fuse_with_offset(LXC_TYPE_PROC_MEMINFO_PATH,
                                                  buf, size, offset, f);
        case LXC_TYPE_PROC_CPUINFO:
-               if (liblxcfs_functional())
+               if (liblxcfs_functional()) {
+                       // assume personality
                        return proc_cpuinfo_read(buf, size, offset, fi);
+                       // restore original personality
+               }

                return read_file_fuse_with_offset(LXC_TYPE_PROC_CPUINFO_PATH,
                                                  buf, size, offset, f);
```